fossology/fossology
# feat(license): Provide predefined license comments

## Description

This feature will allow admins to add some predefined license comments which can be used by users during clearing with the help of a simple drop down in License Comment window.

Admins can navigate to **Admin >> License Admin >> Standard Comments**.

~Currently admins can add **only 7** comments at max to keep the interface simple.~

### Changes

1. Add new table to hold predefined license comments.
1. Allow users to select these comments with the help of a drop down menu.

## How to test

1. Insert some comments from the **Admin Standard License Comments** page.
1. Clear some files and add the comments.
    1. Try adding comments from the decision table.
    1. Try adding comments from the buik table.
1. Add the comments added from the first step should be visible.
1. Added comments should appear in the reports.

**P.S.:** Whenever a change is done in the comments, the user who changed the comment and timestamp is stored as well. It will help in auditing but is never shown in the UI or returned by any API.
